src/Features/Core/Portable/QuickInfo/QuickInfoSection.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    /// <summary>
+    /// Sections are used to make up a <see cref="QuickInfoItem"/>.
+    /// </summary>
+    public sealed class QuickInfoSection
+    {
+        /// <summary>
+        /// The kind of this section. Use <see cref="QuickInfoSectionKinds"/> for the most common kinds.
+        /// </summary>
+        public string Kind { get; }
+
+        /// <summary>
+        /// The individual tagged parts of this section.
+        /// </summary>
+        public ImmutableArray<TaggedText> TaggedParts { get; }
+
+        private QuickInfoSection(string kind, ImmutableArray<TaggedText> taggedParts)
+        {
+            this.Kind = kind ?? string.Empty;
```

Comment:
What does it mean to be string.Empty? It's just information that is otherwise uncategorizable?

Reply:
By the same token, what does it mean to have an empty `TaggedText`? It's just not a useful QuickInfoSection. Is your suggestion that we should throw in this case?